zipkin-lens/src/components/TracePage/TraceSummary.jsx

```diff
@@ -80,19 +83,34 @@ const TraceSummary = React.memo(({ traceSummary }) => {
   }, [isRootedTrace, rootSpanIndex, traceSummary.spans]);
 
   const shownTree = useMemo(() => {
-    const allHiddenSpanIds = {};
-    rerootedTree.forEach((span) => {
-      if (childrenHiddenSpanIds[span.parentId]) {
-        allHiddenSpanIds[span.spanId] = true;
+    let depth = 0;
+    let skip = false;
+
+    return rerootedTree.reduce((acc, cur) => {
+      if (cur.depth > depth && skip) {
```

Comment:
This logic is pretty complex, can you add some comments? In particular, this propagation of `skip` is confusing me.

Reply:
I simplified this logic.
As you said, `skip` is unnecessary... 😓